### ISSUE TITLE
[NV-1680] 🐛 Bug Report: Failed to Parse Source Maps

### ISSUE THREAD

**Issue by JTravaux** [novuhq/novu#2767]
### 📜 Description

Hundreds of webpack _warnings_ relating to parsing source maps in a fresh `create-react-app` project.

### 👟 Reproduction steps

1. Create new react app with `npx create-react-app my-app --template typescript`
2. Install Notification Center `npm i @novu/notification-center`
3. Place the `<NovuProvider subscriberId="subscriberId" applicationIdentifier="appId"></NovuProvider>` anywhere, with anything (or nothing) as children.
4. Run the app with `npm run start`

### 👍 Expected behavior

I would expect to not have these errors in a fresh _(or any)_ project.

### 👎 Actual Behavior with Screenshots

![image](https://user-images.githubusercontent.com/30296275/218880859-99f48f42-0a01-4853-94a0-241842d4c1e4.png)

### 📃 Provide any additional context for the Bug.

These errors no longer appear when `GENERATE_SOURCEMAP=false`, however, I would like to keep source maps enabled if possible. They are just warnings, so it does not affect the application, just litters the console with warnings (156 in this case)

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

<sub>[NV-1680](https://linear.app/novu/issue/NV-1680/🐛-bug-report-failed-to-parse-source-maps)</sub>

**Comment by oba2311** [comment]
@LetItRock  @scopsy  is this a severe issue with the inapp not working out of the box or I'm missing something?

**Comment by oba2311** [comment]
Let me rephrase - if someone is playing with the package - will they run into those errors right now?

**Comment by oba2311** [comment]
i.e. a new user tried to install for the first time

**Comment by scopsy** [comment]
@JTravaux I have run the reproduction steps and after running `npm run start` I have zero warning logs in my console. We also use create-react-app with sourcemaps on our web service and haven't seen those errors either. 
Could you try maybe replicating this in a stackblitz or similar to eliminate some global config you might have?

**Comment by LetItRock** [comment]
@Omer Ben-Ami 

These are not the errors but the warnings, which are only shown during the development but not in the production.

`i.e. a new user tried to install for the first time`

Unfortunately, they will be visible to all the users during the development. 

@Dima Grossman I think I know what might be the problem here, the source maps are pointing to the TS files but should be pointing to the JS files. I feel like the problem might be related to that we have types in the separate `types` folder in the package. We should try to generate types directly to the `esm/cjs` folders.

**Comment by scopsy** [comment]
@LetItRock did you managed to replicate this? I actually couldn't when trying. Also in dev mode

**Comment by LetItRock** [comment]
@scopsy I remember that I saw it some time ago, but I'll try to reproduce it.

**Comment by LetItRock** [comment]
@scopsy I don't see the warnings anymore, but the root problem with the source maps is still there.
You can open dev tools and in the Sources tab search for the `@novu/notification-center` package and then open any component, then you will see that the source maps are not working.

**Comment by scopsy** [comment]
Got ya, so could you please create a new task for the proposed changed so we can track it there?

**Comment by LetItRock** [comment]
the ticket: [https://linear.app/novu/issue/NV-1692/notification-center-source-maps-are-not-working](https://linear.app/novu/issue/NV-1692/notification-center-source-maps-are-not-working)

**Comment by scopsy** [comment]
@JTravaux will keep tracking this in #2810 Since we were unable to replicate this particular error, I believe the fixing #2810 will also resolve your usecase so we will just keep track over the progress there 🙏
